### DISCUSSION TITLE
How to enable logging when running with `--collect-only`

### DISCUSSION THREAD

**Discussion by jacek-lewandowski** [pytest-dev/pytest#8271] [Q&A]
Hi,

I have some custom filtering depending on command line parameters. Some tests are skipped due to something and I want to log the information about why something was skipped. I use standard logging mechanism and run `pytest` with `--collect-only -q`. I've set `log_print=True`, `log_level=INFO`, `log_format` and `log_file`. Although the log file is created, it remains empty after execution. I know that my custom filtering is working fine and I'm sure I go through the code which is expected to log some messages.

Is it possible to enable logging in this case or I'm doing something wrong?

Thanks,
Jacek

**Comment by portyu9** [comment]
Hi @jacek-lewandowski 

This old limitation no longer applies to current pytest.

Modern pytest enables its logging handlers around the **collection phase itself**, so logs emitted while collecting tests can be written to the configured log file even with `--collect-only`.

For example:

```bash
pytest --collect-only \
  --log-file=collection.log \
  --log-file-level=INFO
```

or:

```ini
[pytest]
log_file = collection.log
log_file_level = INFO
```

For live terminal output during collection you can also use:

```bash
pytest --collect-only --log-cli-level=INFO
```

The old `log_print` setting shown in the thread is not a current pytest logging option.

If a current pytest run still produces an empty file, check whether the messages are being emitted through Python's standard `logging` module, whether the logger's effective level permits them, and whether `propagate=False` or a custom handler is preventing the records from reaching pytest.

So today the supported solution is simply the normal `log_file` / `--log-file` configuration; upgrading pytest removes the historical collection-phase limitation.

Please mark this answer as accepted if it helped, thank you.

**Reply by jacek-lewandowski** [reply]
Aww man, it was 6 years ago...

**Reply by portyu9** [reply]
🫥 Im sorry to be the only reply and late but it may help others and maybe if you get back to revisiting it. Dont close it as it may help others if they are searching for the same answer today the supported solution is simply the normal log_file / --log-file configuration; upgrading pytest removes the historical collection-phase limitation.

Please mark this answer as accepted if it helped, thank you.
